linalg/ode.hpp

```diff
@@ -12,20 +12,59 @@
 #ifndef MFEM_ODE
 #define MFEM_ODE
 
+#include "../general/communication.hpp"
 #include "../config/config.hpp"
 #include "operator.hpp"
-#include "../general/communication.hpp"
+#include <vector>
+
 
 namespace mfem
 {
 
+/// A class for storing states of previous timesteps
+class StateData
+{
+protected:
+   MemoryType mem_type;
+   int ss, smax;
+   std::vector<Vector> k;
+   Array<int> idx;
```

Comment:
Why not make these private?  I don't see any derived classes that need the scope of these variables to extend beyond private?